src/lem/circuit.rs

```diff
@@ -998,7 +1161,8 @@ impl Func {
                     // Now we need to enforce that at exactly one path was taken. We do that by enforcing
                     // that the sum of the previously collected `Boolean`s is one. But, of course, this
                     // irrelevant if we're on a virtual path and thus we use an implication gadget.
-                    selector.push(not_dummy.not());
+                    let dummy = not_dummy.not();
+                    selector.push(dummy);
```

Comment:
This isn't immediately clear to me. Shouldn't the selector contain exactly one true value, and shouldn't that value come from the single case in which this is *not* a dummy. I don't understand why you are inverting the boolean here.